### PR TITLE
Tsk987 spotbugs

### PR DESCRIPTION
**Pull request by benjamineckstein**
Last PR of this kind for a while. Made some refactorings to fix more bugs found by Spotbugs.

### CODE REVIEW

File: lib/taskana-core/src/test/java/pro/taskana/impl/integration/TaskServiceImplIntExplicitTest.java [hunk 1]
```diff
@@ -398,6 +398,7 @@ void shouldNotTransferAnyTask() throws SQLException {
       userName = "User",
       groupNames = {"businessadmin"})
   @Test
+  @SuppressWarnings("checkstyle:VariableDeclarationUsageDistance")
```
Comment (mustaphazorgati) [inline lib/taskana-core/src/test/java/pro/taskana/impl/integration/TaskServiceImplIntExplicitTest.java]: why not making them final?
We have checkstyle for a reason :P

Comment (benjamineckstein) [reply lib/taskana-core/src/test/java/pro/taskana/impl/integration/TaskServiceImplIntExplicitTest.java]: good catch! done

**Comment by mustaphazorgati** [comment]
@benjamineckstein did something change? I approved the changes earlier today when you fixed the supresswarning message. 
Afterwards you did force push the branch again

**Comment by benjamineckstein** [comment]
yes because travis was not working and i could not restart the build job without a new commit. 
now your approval is gone :D

**Pull request merged**